presto-main/src/main/java/io/prestosql/sql/rewrite/ShowStatsRewrite.java

```diff
@@ -172,8 +182,15 @@ private Node rewriteShowStats(ShowStats node, Table table, List<SelectItem> sele
             TableMetadata tableMetadata = metadata.getTableMetadata(session, getTableHandle(node, table.getName()));
             Map<String, ColumnHandle> columnHandles = metadata.getColumnHandles(session, tableHandle);
             Set<String> extractNamedResultColumns = extractStatsColumns(tableMetadata, selectItems);
-            List<Expression> resultRows = buildStatisticsRows(tableMetadata, columnHandles, extractNamedResultColumns, tableStatistics);
 
+            try {
```

Comment:
I think this should be moved to `rewriteShowStats` so we check access for `SHOW STATS FOR nation` as well.

Reply:
This code is in the rewriteShowStats method